pulsar-broker/src/test/java/org/apache/pulsar/client/api/TlsProducerConsumerBase.java

```diff
@@ -61,24 +67,42 @@ protected void internalSetUpForBroker() throws Exception {
         conf.setTlsEnabled(true);
         conf.setTlsCertificateFilePath(TLS_SERVER_CERT_FILE_PATH);
         conf.setTlsKeyFilePath(TLS_SERVER_KEY_FILE_PATH);
+        conf.setTlsTrustCertsFilePath(TLS_TRUST_CERT_FILE_PATH);
         conf.setClusterName(clusterName);
+        conf.setTlsRequireTrustedClientCertOnConnect(true);
+        Set<String> tlsProtocols = Sets.newConcurrentHashSet();
+        tlsProtocols.add("TLSv1.2");
+        conf.setTlsProtocols(tlsProtocols);
     }
 
-    protected void internalSetUpForClient() throws Exception {
-        String lookupUrl = new URI("pulsar+ssl://localhost:" + BROKER_PORT_TLS).toString();
-        pulsarClient = PulsarClient.builder().serviceUrl(lookupUrl).tlsTrustCertsFilePath(TLS_SERVER_CERT_FILE_PATH)
-                .enableTls(true).build();
+    protected void internalSetUpForClient(boolean addCertificates, String lookupUrl) throws Exception {
+        ClientConfiguration clientConf = new ClientConfiguration();
+        clientConf.setTlsTrustCertsFilePath(TLS_TRUST_CERT_FILE_PATH);
+        clientConf.setUseTls(true);
+        clientConf.setTlsAllowInsecureConnection(false);
+        if (addCertificates) {
+            Map<String, String> authParams = new HashMap<>();
+            authParams.put("tlsCertFile", TLS_CLIENT_CERT_FILE_PATH);
+            authParams.put("tlsKeyFile", TLS_CLIENT_KEY_FILE_PATH);
+            clientConf.setAuthentication(AuthenticationTls.class.getName(), authParams);
```

Comment:
Do we need TLS *authentication* here?

Reply:
Setting certs is fine, but why authentication is needed?

Reply:
I am a bit confused. 

Without AuthenticationTls how else would you send client side certs to broker?

ConnectionPool.java:90

Reply:
Ah, I may get the point. We need to use AuthenticationTls to SEND client certs whenever we enable `requireTrustedClientCertificate`, even if we wanted to authenticate users and get a role name in other ways (e.g. Basic, Kerberos) on super secure connections, right? And actually we cannot do that (two factor authentication). Is my understanding correct?

If so, using AuthenticationTls makes sense.

Reply:
> We need to use AuthenticationTls to SEND client certs whenever we enable requireTrustedClientCertificate

If TLS is enabled and we want to authenticate using the CNAME in the Client Certs (Default TLS authentication that pulsar provides) then we need to use AuthenticationTls irrespective of requireTrustedClientCertificate 

If we want to use a different authentication method where the connection is TLS but we don't use client certs to authenticate then `requireTrustedClientCertificate` should be false.

Reply:
> If TLS is enabled and we want to authenticate using the CNAME in the Client Certs (Default TLS authentication that pulsar provides) then we need to use AuthenticationTls irrespective of requireTrustedClientCertificate

Correct.

> If we want to use a different authentication method where the connection is TLS but we don't use client certs to authenticate then requireTrustedClientCertificate should be false.

I’m OK with this, however, it depends on usecases. IMO, using both client certs for host authentication and a pair of usename and password for user authentication is a valid usecase.

If we don’t allow to use requireTrustedClientCertificate without AuthenticationTls plugin, shouldn’t it be a configuration for the plugin but not for pulsar global?